tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/Process/ProcessHelper.cs

```diff
@@ -84,22 +129,39 @@ public ProcessResult RunProcess(string command, string[] args, string workingDir
                     }
                 };
 
-                logger.LogDebug($"Running command: {command} {string.Join(" ", args)} in {workingDirectory}");
+                logger.LogDebug("Running command: {command} {args} in {workingDirectory}", command, string.Join(" ", args), workingDirectory);
 
                 process.Start();
                 process.BeginOutputReadLine();
                 process.BeginErrorReadLine();
 
-                process.WaitForExit(processTimeout);
-                exitCode = process.ExitCode;
+                try
+                {
+                    await process.WaitForExitAsync(linkedCts.Token);
+                }
+                catch (OperationCanceledException) when (timeoutCts.IsCancellationRequested)
+                {
+                    output.AppendLine();
+                    output.AppendLine($"Process '{command}' timed out after {timeout.TotalMilliseconds}ms");
+                    logger.LogError("Process '{command}' timed out after {timeout}ms", command, timeout.TotalMilliseconds);
+                    return new ProcessResult { Output = output.ToString(), ExitCode = 124 };
+                }
+                catch (Exception ex)
+                {
+                    output.AppendLine();
+                    output.AppendLine($"Process '{command}' failed with exception: {ex.Message}");
```

Comment:
We should document that we're adding in this output - ie, the user shouldn't expect the command's output to be _only_ what the shell command outputted, on failure.